contributing/metadata.md

```diff
@@ -0,0 +1,173 @@
+
+> Note: The script depends on being able to create an org that will have your type. If you have a working org, but your type isn't compatible with scratch org, you can [use an existing org](#use-an-existing-org)
+
+The script will:
+
+1. Look for missing types (similar to the completeness test)
+2. For missing types, generate a project and scratch org that includes the Features/Settings
+3. Run `force:mdapi:describemetadata` to get the describe
+4. Modify the registry to include the newly found types
+
+```shell
+yarn update-registry
+```
+
+### What the script can't do
+
+inFolderTypes and types with childXml in their describe are not supported. You **want** to explore the various strategies for those see the [SDR Handbook](../HANDBOOK.md) in this repo) and [create tests](#integration-testing) to validate that your types are being handled correctly.
+
+For those situations, refer to another existing type in the registry that you want yours to behave like.
+
+If that's confusing, it's a great time to reach out to the CLI team.
+
+## Manual Edits
+
+You can do what the script does yourself. As you work, run `./node_modules/mocha/bin/mocha test/registry/registryValidation.test.ts` to check your entries
+
+Run `sfdx force:mdapi:describemetadata --json` to get the describe. `>` the output to a file or pipe it to [jq](https://stedolan.github.io/jq/) (`| jq`) to find your type.
+
+Your describe will contains something like this
+
+```json
+{
+    "directoryName": "yourTypes",
+    "inFolder": "false",
+    "metaFile": "false",
+    "suffix": "yourtype",
+    "xmlName": "YourNewType"
+},
+```
+
+Entries in the `describe.json` have the following shape:
+| Property | Description|
+| --- | --- |
+|childXmlNames | List of child sub-components for this component|
+|directoryName | The name of the directory in the .zip file that contains this component.|
+| inFolder | Indicates whether the component is in a folder (true) or not (false). For example, documents, email templates and reports are stored in folders.|
+|metaFile| Indicates whether the component requires an accompanying metadata file. For example, documents, classes, and s-controls are components that require an additional metadata file.|
+|suffix| The file suffix for this component.|
+|xmlName| The name of the root element in the metadata file for this component. This name also appears in the Packages > types > name field in the manifest file package.xml.|
+
+Create an entry in the `types` property:
+
+> this fictitious type COULD HAVE used the automated script
+
+```json
+"yournewtype": { // the key is lowercased xmlName
+      "id": "yournewtype", // lowercased xmlName
+      "name": "YourNewType", // xmlName
+      "suffix": "yourtype", // suffix
+      "directoryName": "yourTypes", //directoryName
+      "inFolder": false, // inFolder
+      "strictDirectoryName": false // set to true when suffixes are not unique in the registry
+    },
+```
```

Comment:
Indent on the last `}` is off. Also I think using `js` syntax would make those comment "errors" less jarring. 
```js
"yournewtype": { // the key is lowercased xmlName
      "id": "yournewtype", // lowercased xmlName
      "name": "YourNewType", // xmlName
      "suffix": "yourtype", // suffix
      "directoryName": "yourTypes", //directoryName
      "inFolder": false, // inFolder
      "strictDirectoryName": false // set to true when suffixes are not unique in the registry
},
```